src/integrationTest/java/com/mongodb/hibernate/IdentifierIntegrationTests.java

```diff
@@ -0,0 +1,250 @@
+        sessionFactoryScope.inTransaction(session -> session.get(InSingleQuotesMixedCase.class, item.id));
+    }
+
+    @Test
+    void startingWithSingleQuote() {
+        var item = new StartingWithSingleQuote();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionStartingWithSingleQuote.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(StartingWithSingleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(StartingWithSingleQuote.class, item.id));
+    }
+
+    @Test
+    void endingWithSingleQuote() {
+        var item = new EndingWithSingleQuote();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionEndingWithSingleQuote.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(EndingWithSingleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(EndingWithSingleQuote.class, item.id));
+    }
+
+    @Test
+    void inDoubleQuotesMixedCase() {
+        var item = new InDoubleQuotesMixedCase();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionInDoubleQuotesMixedCase.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(InDoubleQuotesMixedCase.ACTUAL_FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(InDoubleQuotesMixedCase.class, item.id));
+    }
+
+    @Test
+    void startingWithDoubleQuote() {
+        var item = new StartingWithDoubleQuote();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionStartingWithDoubleQuote.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(StartingWithDoubleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(StartingWithDoubleQuote.class, item.id));
+    }
+
+    @Test
+    void endingWithDoubleQuote() {
+        var item = new EndingWithDoubleQuote();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionEndingWithDoubleQuote.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(EndingWithDoubleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(EndingWithDoubleQuote.class, item.id));
```

Comment:
the above testing cases seem to follow similar pattern and is paramerized by two inputs: 1) entity class; 2) identifier name supplier, so it is doable to avoid the duplication by JUnit 5's feature.

but this is optional and over-abstracting in testing code not necessarily helps on code maintaining.

Reply:
I don't expect there to be a meaningful improvement if we try and replace the current simple code by extracting the common code and introducing parameters. Given that you marked this thread as optional, I am resolving it (the PR author, despite not being able to resolve other conversations started by a reviewer, is free to resolve optional conversations - that's what optional comments are).